node-api/backend/validator.go

```diff
@@ -228,21 +242,49 @@ func (b Backend) ValidatorByID(
 func (b Backend) ValidatorBalancesByIDs(
 	slot math.Slot, ids []string,
 ) ([]*beacontypes.ValidatorBalanceData, error) {
-	var index math.U64
 	st, _, err := b.stateFromSlot(slot)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to get state from slot %d", slot)
 	}
+
+	// If no IDs provided, return all validator balances
```

Comment:
what are the specs here?
We return every validator balance or only active ones?
Note that a validator entry in create in beaconState the moment a deposit is successfully processed, even if it has 10K and it cannot be activated yet.

Reply:
`Balances will be returned for all indices or public key that match known validators.If an index or public key does not match any known validator, no balance will be returned but this will not cause an error.` from [reference](https://ethereum.github.io/beacon-APIs/#/Beacon/getStateValidatorBalances).
Here status is not a filter, so all validator balance matching the filter.